tests/test-grammar-integration.cpp

```diff
@@ -231,13 +319,15 @@ number ::= [0-9]+)""";
     // Ensure we did NOT parsed correctly
     assert(parsed_grammar.rules.empty());
 
-    fprintf(stderr, "End of expected error. Test successful.\n");
```

Comment:
Hah, I found that extra reassurance useful to scan through "errors" tbh

Reply:
Because I was saying "Test successful" here, it was ambiguous if I meant this particular test, or the entire suite of tests (for instance, this particular `test_failure_missing_root` could have passed, but a later test failed, and it could be confusing if that behavior was expected).

So now I'm still printing a line like this -- I just moved printing "All tests passed." to the very end, so that I only print this part if indeed all tests were successful. 

By breaking it up onto separate lines, I was hoping to make things more explicit.